lib/hanami/container.rb

```diff
@@ -0,0 +1,115 @@
+# frozen_string_literal: true
+
+require "dry/system/container"
+
+module Hanami
+  # Hanami private IoC
+  #
+  # @since 2.0.0
+  class Container < Dry::System::Container
+    boot(:root) do
```

Comment:
Another idea would be to have a whole framework-specific _container_, which then gets imported from application and slice containers, this would take care of the namespacing and also help encourage better separation of the code within the various broad sections of the app

Reply:
Hmm. @timriley, you give me an idea. I think, it will be awesome to put `Hanami.boot` to general container and generate `./system/` folder as a part of framework. And in this case instead this code:
```
# frozen_string_literal: true

require_relative './environment'
Hanami.boot
```

use something like this:

```
Hanami::Container.boot!
```

And also, anyone will can use dry-system booting process instead `./config/initializers/*.rb` magic 🤔

Reply:
oh yes, we should be 100% offering dry-system bootable components instead of random initialisers!

Reply:
@davydovanton perhaps you could demonstrate what this change would look like, perhaps a diff based on this PR?

Reply:
@davydovanton Given we're not sure if we're gonna have a separate place for settings, or multiple containers, I'd leave this as it is.

We need to address this aspect soon.

Reply:
@jodosha agree, I can create proposal and we can start discussion. WDYT?